Sources/Commands/SwiftTool.swift

```diff
@@ -267,12 +267,17 @@ public class SwiftTool<Options: ToolOptions> {
             verbosity = Verbosity(rawValue: options.verbosity)
             Process.verbose = verbosity != .concise
             // Call the implementation.
-            try runImpl()
+            let status = try runImpl()
             if diagnostics.hasErrors {
                 throw Error.hasFatalDiagnostics
             }
             // Print any non fatal diagnostics like warnings, notes.
             printDiagnostics()
+
+            switch status {
+            case .success: POSIX.exit(0)
+            case let .failure(code): POSIX.exit(Int32(code))
+            }
         } catch {
             printDiagnostics()
             handle(error: error)
```

Comment:
We should remove the `exit(1)` from this method and move the above switch statement after the `try..catch`.